R/causal_impact.R

```
@@ -95,6 +96,22 @@ do_causal_impact_ <- function(df, time_col, formula, intervention_time = NULL, o
     input_df <- move_col(input_df, y_colname, 1)
 
     df_zoo <- zoo::zoo(input_df, time_points_vec)
+    # fill NAs in the input.
+    if (na_fill_type == "spline") {
+      df_zoo <- zoo::na.spline(df_zoo)
+    }
+    else if (na_fill_type == "interpolate") {
+      df_zoo <- zoo::na.approx(df_zoo)
+    }
+    else if (na_fill_type == "locf") {
+      df_zoo <- zoo::na.locf(df_zoo)
+    }
+    else if (na_fill_type == "StructTS") {
+      df_zoo <- zoo::na.StructTS(df_zoo)
+    }
+    else if (na_fill_type == "value") {
+      df_zoo <- zoo::na.fill(na_fill_value)
+    }
```

Or maybe it should throw an error